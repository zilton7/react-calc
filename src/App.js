function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h3>React Calc</h3>
      </header>
    </div>
  );
}

export default App;
