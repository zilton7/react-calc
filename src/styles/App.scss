html {
  font-size: 62.5%;
}

body {
  display: flex;
  justify-content: center;
  align-items: center;
}

.calculator {
  width: 70rem;
}

button {
  width: 17.5rem;
}

.group {
  display: flex;
  flex-direction: row;
}

.button-panel {
  margin: 2rem auto;
  width: 100%;
  text-align: center;
}

input#result {
  text-align: right;
  width: 100%;
  margin: 0 1rem;
  background: rgb(177, 177, 177);
  color: white;
}

.display {
  display: flex;
  justify-content: center;
}
