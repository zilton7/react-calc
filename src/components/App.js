import React, { Component } from "react";
import { useState } from "react";
import Display from "./Display";
import ButtonPanel from "./ButtonPanel";
import "../styles/App.scss";
import calculate from "../logic/calculate";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick.bind(this);
  }

  handleClick = (buttonName) => {
    const data = this.state;
    const result = calculate(data, buttonName);
    this.setState({
      total: result.total,
      next: result.next,
      operation: result.operation,
    });
  };

  render() {
    return (
      <>
        <h1>Calculator</h1>
        <Display calculate={calculate} />
        <ButtonPanel />
      </>
    );
  }
}

export default App;
