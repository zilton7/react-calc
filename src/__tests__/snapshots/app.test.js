import renderer from "react-test-renderer";
import App from "../../components/App";

it("renders App correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
