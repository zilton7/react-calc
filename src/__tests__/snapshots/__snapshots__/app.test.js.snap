// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<main>
  <div
    className="navbar"
  >
    <h3>
      Math Magicians
    </h3>
    <div
      className="navbar-links"
    >
      <a
        className="navbar-link"
        href="/"
      >
        Home
      </a>
      <a
        className="navbar-link"
        href="/calculator"
      >
        Calculator
      </a>
      <a
        className="navbar-link"
        href="/quote"
      >
        Quote
      </a>
    </div>
  </div>
  <h2>
    Welcome to our page!
  </h2>
  <p
    id="article"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
</main>
`;
